src/CMakeLists.txt

```diff
@@ -63,8 +63,27 @@ if(NOT CMAKE_BUILD_TYPE)
           FORCE
   )
 endif()
+if(CMAKE_BUILD_TYPE MATCHES "Release")
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    # Clang has some issues with thinLTO (segfaults on final linking)
+    # This enables full LTO through CMake
+    # FullLTO is slow, but may yield performance improvements.
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -flto")
```

Comment:
Do we want to revisit this at some time to see if performance is improved?

Reply:
I can confirm it does fix a bug where LTO wasn't being properly set on Emscriptem 2.X
The thing I want to ask @jturner65 is what the default config should be. LTO can slow things down a lot, but it is almost guaranteed to perform a minor performance improvement (particularly with the cyclic dependencies we have in our Codebase now). 

Should calling setup.py install a nice a debug friendly Release? Or should it install one optimized as best as possible. We can try to do both, but I think GCC may have some issue with symbols mangling with LTO enabled for GCC <= 7. Not to mention it does slow linking quite a bit on my underpowered Mac.